server/service/src/dashboard/item_count.rs

```diff
@@ -56,6 +61,30 @@ pub trait ItemCountServiceTrait: Send + Sync {
 
         return more_than_six_mos_count;
     }
+
+    fn get_out_of_stock_products_count(&self, item_stats: &Vec<ItemStats>) -> i64 {
+        let out_of_stock_products_count = item_stats
+            .iter()
+            .filter(|i| i.total_consumption > 0.0 && i.total_stock_on_hand == 0.0)
+            .count() as i64;
+
+        return out_of_stock_products_count;
```

Comment:
Nit: 

```suggestion
        item_stats
            .iter()
            .filter(|i| i.total_consumption > 0.0 && i.total_stock_on_hand == 0.0)
            .count() as i64
```

down below too please